frostsnapp/native/src/coordinator.rs

```diff
@@ -9,97 +8,81 @@ use frostsnap_coordinator::firmware_upgrade::{
+use frostsnap_core::{
+    coordinator::{FrostCoordinator, SigningSessionState},
+    DeviceId, KeyId, SignTask,
+};
 use std::collections::{BTreeSet, VecDeque};
-use std::fs::File;
 use std::sync::{Arc, Mutex};
 use std::thread::JoinHandle;
 use std::time::Duration;
 use tracing::{event, Level};
 
 pub struct FfiCoordinator {
-    coordinator: Arc<Mutex<FrostCoordinator>>,
     usb_manager: Mutex<Option<UsbSerialManager>>,
     pending_for_outbox: Arc<Mutex<VecDeque<CoordinatorSend>>>,
     key_event_stream: Arc<Mutex<Option<StreamSink<KeyState>>>>,
     thread_handle: Mutex<Option<JoinHandle<()>>>,
     ui_protocol: Arc<Mutex<Option<Box<dyn UiProtocol>>>>,
-    db: Arc<Mutex<LlsDb<File>>>,
-    persist_core: IndexHandle<PersistCore>,
-    device_names: IndexHandle<llsdb::index::BTreeMap<DeviceId, String>>,
     usb_sender: UsbSender,
     firmware_bin: FirmwareBin,
     firmware_upgrade_progress: Arc<Mutex<Option<StreamSink<f32>>>>,
+
+    // persisted things
+    db: Arc<Mutex<rusqlite::Connection>>,
+    device_names: Arc<Mutex<Persisted<DeviceNames>>>,
+    coordinator: Arc<Mutex<Persisted<FrostCoordinator>>>,
+    signing_session: Arc<Mutex<Persisted<Option<SigningSessionState>>>>,
 }
 
 impl FfiCoordinator {
     pub fn new(
-        db: Arc<Mutex<LlsDb<File>>>,
-        mut usb_manager: UsbSerialManager,
+        db: Arc<Mutex<rusqlite::Connection>>,
+        usb_manager: UsbSerialManager,
     ) -> anyhow::Result<Self> {
         let pending_for_outbox = Arc::new(Mutex::new(VecDeque::new()));
 
-        let (persist_core, device_names_handle, coordinator) = db
-            .lock()
-            .unwrap()
-            .execute(|tx| {
-                let persist = PersistCore::new(tx)?;
-                let handle = tx.store_index(persist);
-                let device_names_list = tx.take_list("device_names")?;
-                let device_names_index = llsdb::index::BTreeMap::new(device_names_list, &tx)?;
-                let device_names_handle = tx.store_index(device_names_index);
-                let core_api = tx.take_index(handle);
-                let coordinator = core_api.core_coordinator()?;
-                let device_names = tx.take_index(device_names_handle);
-                *usb_manager.device_labels_mut() = device_names
-                    .iter()
-                    .collect::<Result<_>>()
-                    .context("reading in device names from disk")?;
-
-                Ok((handle, device_names_handle, coordinator))
-            })
-            .context("initializing db")?;
+        let mut db_ = db.lock().unwrap();
+
+        let coordinator = Persisted::<FrostCoordinator>::new(&mut db_, ())?;
+        let device_names = Persisted::<DeviceNames>::new(&mut db_, ())?;
+        let signing_session = Persisted::<Option<SigningSessionState>>::new(&mut db_, ())?;
 
         let usb_sender = usb_manager.usb_sender();
 
         // HACK: if the global device list depends on db state then it shouldn't be global! The
```

Comment:
nit: This is solved now? Can remove this comment

Reply:
Thanks removed.